src/pages/home/sidebar/SidebarLinksData.js

```diff
@@ -134,55 +55,23 @@ function SidebarLinksData({
     // eslint-disable-next-line react-hooks/exhaustive-deps
     useEffect(() => Policy.openWorkspace(activeWorkspaceID, policyMemberAccountIDs), [activeWorkspaceID]);
 
-    const reportIDsRef = useRef(null);
+    const orderedReportIDsRef = useRef(null);
```

Comment:
*question:* Is this ref still needed? Since `useReportIDs` just uses `useContext`, and we know that that context value is a stable reference, `orderedReportIDs` should also be a stable reference, right? and if that's the case, then we should be able to simplify this:

```diff
diff --git a/src/pages/home/sidebar/SidebarLinksData.js b/src/pages/home/sidebar/SidebarLinksData.js
index 35c4a2c59c..384b2c6062 100644
--- a/src/pages/home/sidebar/SidebarLinksData.js
+++ b/src/pages/home/sidebar/SidebarLinksData.js
@@ -1,16 +1,13 @@
-import {deepEqual} from 'fast-equals';
 import PropTypes from 'prop-types';
-import React, {useCallback, useEffect, useMemo, useRef} from 'react';
+import React, {useCallback, useEffect, useRef} from 'react';
 import {View} from 'react-native';
 import {withOnyx} from 'react-native-onyx';
 import networkPropTypes from '@components/networkPropTypes';
-import {withNetwork} from '@components/OnyxProvider';
 import withCurrentReportID from '@components/withCurrentReportID';
 import withCurrentUserPersonalDetails, {withCurrentUserPersonalDetailsDefaultProps, withCurrentUserPersonalDetailsPropTypes} from '@components/withCurrentUserPersonalDetails';
 import withNavigationFocus from '@components/withNavigationFocus';
 import useActiveWorkspace from '@hooks/useActiveWorkspace';
 import useLocalize from '@hooks/useLocalize';
-import usePrevious from '@hooks/usePrevious';
 import {useReportIDs} from '@hooks/useReportIDs';
 import useThemeStyles from '@hooks/useThemeStyles';
 import compose from '@libs/compose';
@@ -44,35 +41,19 @@ const defaultProps = {
     ...withCurrentUserPersonalDetailsDefaultProps,
 };
 
-function SidebarLinksData({isFocused, currentReportID, insets, isLoadingApp, onLinkClick, priorityMode, network, policyMembers, currentUserPersonalDetails}) {
+function SidebarLinksData({isFocused, currentReportID, insets, isLoadingApp, onLinkClick, priorityMode, policyMembers, currentUserPersonalDetails}) {
     const styles = useThemeStyles();
     const {activeWorkspaceID} = useActiveWorkspace();
     const {translate} = useLocalize();
-    const prevPriorityMode = usePrevious(priorityMode);
 
     const policyMemberAccountIDs = getPolicyMembersByIdWithoutCurrentUser(policyMembers, activeWorkspaceID, currentUserPersonalDetails.accountID);
 
     // eslint-disable-next-line react-hooks/exhaustive-deps
     useEffect(() => Policy.openWorkspace(activeWorkspaceID, policyMemberAccountIDs), [activeWorkspaceID]);
 
-    const orderedReportIDsRef = useRef(null);
     const isLoading = isLoadingApp;
     const {orderedReportIDs} = useReportIDs();
 
-    const optionListItems = useMemo(() => {
-        if (deepEqual(orderedReportIDsRef.current, orderedReportIDs)) {
-            return orderedReportIDsRef.current;
-        }
-
-        // 1. We need to update existing reports only once while loading because they are updated several times during loading and causes this regression: https://github.com/Expensify/App/issues/24596#issuecomment-1681679531
-        // 2. If the user is offline, we need to update the reports unconditionally, since the loading of report data might be stuck in this case.
-        // 3. Changing priority mode to Most Recent will call OpenApp. If there is an existing reports and the priority mode is updated, we want to immediately update the list instead of waiting the OpenApp request to complete
-        if (!isLoading || !orderedReportIDsRef.current || network.isOffline || (orderedReportIDsRef.current && prevPriorityMode !== priorityMode)) {
-            orderedReportIDsRef.current = orderedReportIDs;
-        }
-        return orderedReportIDsRef.current || [];
-    }, [orderedReportIDs, isLoading, network.isOffline, prevPriorityMode, priorityMode]);
-
     const currentReportIDRef = useRef(currentReportID);
     currentReportIDRef.current = currentReportID;
     const isActiveReport = useCallback((reportID) => currentReportIDRef.current === reportID, []);
@@ -92,7 +73,7 @@ function SidebarLinksData({isFocused, currentReportID, insets, isLoadingApp, onL
                 isActiveReport={isActiveReport}
                 isLoading={isLoading}
                 activeWorkspaceID={activeWorkspaceID}
-                optionListItems={optionListItems}
+                optionListItems={orderedReportIDs}
             />
         </View>
     );
@@ -106,7 +87,6 @@ export default compose(
     withCurrentReportID,
     withCurrentUserPersonalDetails,
     withNavigationFocus,
-    withNetwork(),
     withOnyx({
         isLoadingApp: {
             key: ONYXKEYS.IS_LOADING_APP,

```

Reply:
I agree. I only kept this around because of the comments it had but then paying attention to the implementation, this ref is not adding any value. I also tested the linked regression in the comments and it doesn't happen now, even after removing the ref.

So I think we are safe to remove it.